### PR TITLE
User endpoints

### LINKED ISSUE #55: Implement Shoutout Server User endpoints
As a developer, I would like to have a shoutout endpoint for: 

- [x] An array of all users in the database with:
    - First Name
    - Last Name
    - Slack Handle
    - Total shouts given
    - Total shouts received

- [x] An array of a specific user's shoutouts **_given_** with in the last 30 days by default
- [x] An array of a specific user's shoutouts **_received_** with in the last 30 days by default

Estimate: 3h

### PR DESCRIPTION
**Pull request by tgeeTechtonic**
### Type of change:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling - no new features

### Why is this change required? What problem(s) does it solve?
Added 2 endpoints where you can get a list of all users in the database and all shoutouts for a specific user by date range

### Link issues
Closes #55

### CODE REVIEW

File: server/controllers/shoutoutController.js [hunk 1]
```diff
@@ -31,14 +29,24 @@ const shoutoutController = {
...
-      .then((shoutouts) => {
-        const shoutoutPromises = shoutouts
-          .sort((a, b) => b.date - a.date)
-          .map((shout) => createShoutoutRes(db, shout));
-        return Promise.all(shoutoutPromises);
-      })
+      .orderBy('date', 'desc')
+      .then((shouts) =>
+        Promise.all(shouts.map((shout) => createShoutoutRes(db, shout)))
+      )
```
Comment (emackinnonTechtonic) [inline server/controllers/shoutoutController.js:35]: Nice refactor here

**Review by emackinnonTechtonic — APPROVED** [review]

File: server/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  return {
+    id,
+    first_name,
+    last_name,
+    num_shoutouts_given: parseInt(shoutsGiven[0].count),
+    num_shoutouts_received: parseInt(shoutsReceived[0].count),
+    slack_handle,
+  };
+};
```
Comment (broxburyTechtonic) [inline server/controllers/userController.js:29]: If we end up wanting to have an expanding table for shoutouts is this where we would put that?

File: server/controllers/shoutoutController.js [hunk 3]
```diff
@@ -31,14 +29,24 @@ const shoutoutController = {
...
       .then((formattedShoutouts) => formattedShoutouts);
   },
+  getShoutoutsByuserId: async (type, userId, days) => {
+    const lastDay = new Date();
+    const firstDay = new Date(new Date().setDate(lastDay.getDate() - days));
+
+    return await db('shoutouts')
+      .whereBetween('date', [firstDay, lastDay])
+      .where(type, userId)
+      .orderBy('date', 'desc')
```
Comment (broxburyTechtonic) [inline server/controllers/shoutoutController.js:45]: I think the Month picker can be used here for the FE to pass the firstDay and last day of the two selected months. Something to think about as we build out that functionality for the User's table.

**Review by broxburyTechtonic — APPROVED** [review]
Approved!

**Pull request merged**